crates/holochain/src/core/workflow/sys_validation_workflow/unit_tests.rs

```diff
@@ -0,0 +1,393 @@
+        let action_hashed = ActionHashed::from_content_sync(action);
+        SignedActionHashed::sign(&self.keystore, action_hashed)
+            .await
+            .unwrap()
+    }
+
+    fn with_network_behaviour(&mut self, network: MockHolochainP2pDnaT) -> &mut Self {
+        self.actual_network = Some(network);
+        self
+    }
+
+    async fn save_op_to_db<T: DbKindT>(
+        &self,
+        db: DbWrite<T>,
+        op: DhtOp,
+    ) -> StateMutationResult<DhtOpHash> {
+        let op = DhtOpHashed::from_content_sync(op);
+
+        let test_op_hash = op.as_hash().clone();
+        db.write_async({
+            move |txn| -> StateMutationResult<()> {
+                holochain_state::mutations::insert_op(txn, &op)?;
+                Ok(())
+            }
+        })
+        .await
+        .unwrap();
+
+        Ok(test_op_hash)
+    }
+
+    async fn run(&mut self) {
+        // TODO So this struct is just here to follow the 'workspace' pattern? The Space gets passed to the workflow anyway and most of the fields are shared.
+        //      Maybe just moving the Space to the workspace is enough to tidy this up?
+        let workspace = SysValidationWorkspace::new(
+            self.test_space.space.authored_db.clone().into(),
+            self.test_space.space.dht_db.clone().into(),
+            self.test_space.space.dht_query_cache.clone(),
+            self.test_space.space.cache_db.clone().into(),
+            Arc::new(self.dna_def.clone()),
+        );
+
+        let mut network = MockHolochainP2pDnaT::new();
+        // This can't be copied so if you want to call the run twice you would need to reset the network behaviour!
+        let actual_network = self
+            .actual_network
+            .take()
+            .unwrap_or_else(|| MockHolochainP2pDnaT::new());
+        network.expect_clone().return_once(move || actual_network);
+
+        sys_validation_workflow(
+            Arc::new(workspace),
+            Arc::new(self.test_space.space.clone()),
+            self.app_validation_trigger.0.clone(),
+            self.self_trigger.0.clone(),
+            network,
+        )
+        .await
+        .unwrap();
+    }
+
+    /// This provides a quick and reliable way to check that ops have been sys validated
+    async fn get_ops_pending_app_validation(&self) -> HashSet<DhtOpHash> {
+        get_ops_to_app_validate(&self.dht_db_handle().into())
+            .await
+            .unwrap()
+            .into_iter()
+            .map(|op_hashed| op_hashed.hash)
+            .collect()
+    }
+
+    async fn expect_app_validation_triggered(&mut self) {
+        tokio::time::timeout(
+            std::time::Duration::from_secs(3),
+            self.app_validation_trigger.1.listen(),
+        )
+        .await
+        .expect("Timed out waiting for app validation to be triggered")
+        .unwrap();
+    }
+
+    // TODO The app validation workflow is unconditionally triggered
```

Comment:
yeah, it's a relatively minor thing, but the correct thing would be for this to be used :)

Reply:
Yes absolutely, as part of addressing the unhelpful trigger this code will become used